packages/store/ts/library/render-solidity/renderTable.ts

```diff
@@ -23,12 +24,15 @@ export function renderTable(options: RenderTableOptions) {
     fields,
     staticFields,
     dynamicFields,
+    withFieldMethods,
     withRecordMethods,
+    withEphemeralMethods,
     storeArgument,
     primaryKeys,
   } = options;
 
   const { _typedTableId, _typedKeyArgs, _primaryKeysDefinition } = renderCommonData(options);
+  const shouldRenderDelete = !withEphemeralMethods;
```

Comment:
For consistency we should add a `withDeleteMethods` field to the config (like `withFieldMethods`, `withRecordMethods`, `withEphemeralMethods`). Not strictly blocking for this PR, we can also do that in a follow up.